ros2interface/ros2interface/api/__init__.py

```diff
@@ -0,0 +1,150 @@
+    for package_name, service_names in get_all_service_types().items():
+        for service_name in service_names:
+            types.append(
+                '{package_name}/srv/{service_name}'.format_map(locals()))
+
+    for package_name, message_names in get_all_message_types().items():
+        for message_name in message_names:
+            types.append(
+                '{package_name}/msg/{message_name}'.format_map(locals()))
+    return sorted(types)
+
+
+def get_all_action_types():
+    all_action_types = {}
+    for package_name in get_resources('rosidl_interfaces'):
+        action_types = get_action_types(package_name)
+        if action_types:
+            all_action_types[package_name] = action_types
+    return all_action_types
+
+
+def get_action_types(package_name):
+    if not has_resource('packages', package_name):
+        raise LookupError('Unknown package name')
+    try:
+        content, _ = get_resource('rosidl_interfaces', package_name)
+    except LookupError:
+        return []
+    interface_names = content.splitlines()
+    # TODO(jacobperron) this logic should come from a rosidl related package
+    # Only return actions in action folder
+    return list(sorted({
+        n[7:-7]
+        for n in interface_names
+        if n.startswith('action/') and n[-7:] in ('.idl', '.action')}))
+
+
+def get_all_message_types():
+    all_message_types = {}
+    for package_name in get_resources('rosidl_interfaces'):
+        message_types = get_message_types(package_name)
+        if message_types:
+            all_message_types[package_name] = message_types
+    return all_message_types
+
+
+def get_message_types(package_name):
+    if not has_resource('packages', package_name):
+        raise LookupError('Unknown package name')
+    try:
+        content, _ = get_resource('rosidl_interfaces', package_name)
+    except LookupError:
+        return []
+    interface_names = content.splitlines()
+    # TODO(dirk-thomas) this logic should come from a rosidl related package
+    # Only return messages in msg folder
+    return list(sorted({
+        n[4:-4]
+        for n in interface_names
+        if n.startswith('msg/') and n[-4:] in ('.idl', '.msg')}))
+
+
+def get_all_service_types():
+    all_service_types = {}
+    for package_name in get_resources('rosidl_interfaces'):
+        service_types = get_service_types(package_name)
+        if service_types:
+            all_service_types[package_name] = service_types
+    return all_service_types
+
+
+def get_service_types(package_name):
+    if not has_resource('packages', package_name):
+        raise LookupError('Unknown package name')
+    try:
+        content, _ = get_resource('rosidl_interfaces', package_name)
+    except LookupError:
+        return []
+    interface_names = content.splitlines()
+    # TODO(dirk-thomas) this logic should come from a rosidl related package
+    # Only return services in srv folder
+    return list(sorted({
+        n[4:-4]
```

Comment:
AFAICT, wherever we use `get_service_types`, we are re-inserting the the `srv/` namespace.
Unless there is a reason to not include the namespace in the names returned by this function, I would include it.
Same for the other `get_*_types()` functions.